config/areas/site/dialogs.php

```diff
@@ -479,6 +479,26 @@
 				];
 			}
 
+			$slugAppendix  = Str::slug(I18n::translate('page.duplicate.appendix'));
+			$titleAppendix = I18n::translate('page.duplicate.appendix');
+
+			// if the item to be duplicated already exists
+			// add a suffix at the end of slug and title
+			$duplicateSlug = $page->slug() . '-' . $slugAppendix;
+			$siblingKeys   = $page->parentModel()->childrenAndDrafts()->pluck('uid');
+
+			if (in_array($duplicateSlug, $siblingKeys) === true) {
+				$suffixCounter = 2;
+				$newSlug       = $duplicateSlug . $suffixCounter;
+
+				while (in_array($newSlug, $siblingKeys) === true) {
+					$newSlug = $duplicateSlug . ++$suffixCounter;
+				}
+
+				$slugAppendix  .= $suffixCounter;
+				$titleAppendix .= ' ' . $suffixCounter;
+			}
```

Comment:
I think it would be great to add this code as a universal method to  the `Str` class. Not sure about the name but maybe

```php
$siblings   = $page->parentModel()->childrenAndDrafts()->pluck('uid');
$slug = Str::unique($page->slug(), fn ($slug) => in_array($slug, $siblings) === false, $appendix);
```

Reply:
I tried it in a very similar way to yours, but it seemed to me to be a very special method for this case.

Also `Str::unique()` should return the new appendix, not the updated version of the first parameter. Because we need to use appendix for both slug and title.

As a name, it can be `Str::appendix()`.